statsmodels/base/model.py

```diff
@@ -743,34 +743,14 @@ def predict(self, exog=None, transform=True, *args, **kwargs):
             See self.model.predict
 
         """
-        import pandas as pd
+        from statsmodels.tools.data_interface import NumPyInterface
 
-        exog_index = exog.index if _is_using_pandas(exog, None) else None
-
-        if transform and hasattr(self.model, 'formula') and exog is not None:
-            from patsy import dmatrix
-            exog = dmatrix(self.model.data.design_info.builder,
-                           exog)
-
-        if exog is not None:
-            exog = np.asarray(exog)
-            if exog.ndim == 1 and (self.model.exog.ndim == 1 or
-                                   self.model.exog.shape[1] == 1):
```

Comment:
Interface could expose an property that is atleast2d or similar.  Alternatively it could use a keyword to always expose a minimum number of dimensions, e.g.

```
exog_interface = NumPyInterface(data=exog, model=self.model, use_formula=transform, min_dims=2)
```